src/Raven.Server/Documents/Handlers/DocumentReplicationHandler.cs

```diff
@@ -107,6 +107,13 @@ public Task GetLastSentEtag()
                                     }
                                     writer.FinalizeDocument();
                                     var receivedDoc = writer.CreateReader();
+
+                                    //special document that signifies "Get Last Etag" request
+                                    if (receivedDoc.Count == 1 && receivedDoc.GetPropertyNames()[0] == "Raven/GetLastEtag")
```

Comment:
Just for example, if user has a lot of documents with one property name, this is going to allocate a lot of strings, just to compare them to `Raven/GetLastEtag`